pkg/condition/types/types_test.go

```diff
@@ -173,6 +173,20 @@ func TestPrimitives(t *testing.T) {
 			repr:          "ipaddress",
 			expectedError: fmt.Errorf("for ipaddress: could not parse string as an ipaddress `invalid`: ParseAddr(\"invalid\"): unable to parse IP"),
 		},
+		{
+			name:      "valid_map_string",
+			paramType: mustMapParamType(StringParamType),
+			input:     map[string]string{"hello": "world"},
+			output:    map[string]string{"hello": "world"},
+			repr:      "TYPE_NAME_MAP<string>",
+		},
+		{
+			name:      "valid_list_string",
+			paramType: mustListParamType(StringParamType),
+			input:     []string{"hello", "world"},
```

Comment:
Easy fix here, these should be the generic type `[]any`. The converters will ensure the generic types are converted according to the `list<type>` definition.

So this should change to

```go
input: []any{"hello", "world"},
output: []any{"hello", "world"}
```
then we can also add a test to make sure the generic type converter is working as expect. For example,

```go
			name:      "invalid_list_string",
			paramType: mustListParamType(StringParamType),
			input:     []any{"hello", 1},
			expectedErr: ...,
			repr:      "TYPE_NAME_LIST<string>",
		},
```